### PR TITLE
Markdown Spellcheck

### LINKED ISSUE #173: Markdown Linting
If a file is a markdown file, we should lint it for style or maybe some obvious typos.

### PR DESCRIPTION
**Pull request by yo-404**
Resolves #173 

## Description

Changes proposed in this pull request:
- Added Language Tool
- Added Misspell check

Files Added
- spellcheck.yaml @.github\workflows\

### Tested on

forked repo

Results -
https://github.com/yo-404/saferplace/actions/runs/6121405779

### CODE REVIEW

File: package-lock.json
Comment (VoyTechnology) [inline package-lock.json:1]: The project is using pnpm and already has the lock file for that, do you mind removing this file?

File: .github/workflows/spellcheck.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+          enabled_categories: ''
+          patterns: "**.md"
+    
+      - name: running misspell
+        # To perform misspell check even after the language tool test fails 
+        if: success() || failure()  
+        uses: reviewdog/action-misspell@v1
+        with:
+          github_token: ${{ secrets.github_token }}
+          locale: "US"
```
Comment (VoyTechnology) [inline .github/workflows/spellcheck.yml:34]: We will have to convert it to a matrix later to support i18n (#63 as well as #175). We can do it now or add it later, so feel free to dismiss this review if you don't want to do it.

File: .github/workflows/spellcheck.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
+name: Spellcheck
+on: 
+  push:
+    # branches: [main]
```
Comment (VoyTechnology) [inline .github/workflows/spellcheck.yml:4]: We might want to add path filtering, so we only trigger this change on `*.md` file changes. That way we don't run this unnecessarily if we just change some TS or Go files. You can take a look at other workflows as to how its done.

**Review by VoyTechnology — CHANGES_REQUESTED** [review]

File: .github/workflows/spellcheck.yml (hunk 2 above)
Comment (yo-404) [reply .github/workflows/spellcheck.yml:4]: sure ...will look into it and try to add some filters

**Review by VoyTechnology — APPROVED** [review]

**Pull request merged**